### PR TITLE
SN-6607: Add unit test for is_comm_buffer_parse_messages()

### PR DESCRIPTION
**Pull request by tmcd249**
- (SDK) Implimented unit test for is_comm_buffer_parse_messages().
- (SDK) Fixes edge case found while implementing unit test.

### CODE REVIEW

File: tests/test_ISComm.cpp [hunk 1]
```diff
@@ -1319,3 +1318,255 @@ TEST(ISComm, TruncatedPackets)
...
+#define BUFF_PARSE_INS          4  
+
+static uint32_t s_buffParseMsgInCnt[5] = { 0 };
+
+/**
+ * @brief ISB callback for testing the function is_comm_buffer_parse_messages()
+ */
+int BufferParse_isb(unsigned int port, p_data_t* data)
+{
+	switch (data->hdr.id)
```
Comment (kylemallory) [inline tests/test_ISComm.cpp:1340]: TABS!!!

File: tests/test_ISComm.cpp [hunk 2]
```diff
@@ -1319,3 +1318,255 @@ TEST(ISComm, TruncatedPackets)
...
+TEST(ISComm, BufferParse)
+{
+    is_comm_instance_t comm;
+
+    is_comm_callbacks_t callbacks = {};
+	callbacks.isbData = BufferParse_isb;
+	callbacks.nmea = BufferParse_nmea;
+
+    uint32_t msgOutCnt[5] = {0};
+    uint8_t randomBuf[16] = {0xe0,0x63,0xa4,0x17,0x95,0xb1,0x26,0xd2,0xe0,0x63,0xa4,0x17,0x95,0xb1,0x26,0xd2};
```
Comment (kylemallory) [inline tests/test_ISComm.cpp:1375]: Is this random data?  If so, it should probably be randomly generated (dynamically).  Also, a lot of the following code appears to be "test data".  I think we have some routines that are responsible for actually generating test data, with some random parameters, etc.

At best, you should be trying to use those functions. Otherwise, we should try and make this reuseable, so that if we need custom data like this elsewhere, it can be reused.

**Review by kylemallory — CHANGES_REQUESTED** [review]
There are a lot of mixed whitespace in the new code.  Let's clean that up.

Also, there are a number of existing mechanisms in test_ISComm that are supposed to test other aspects of ISComm.  I guess I'm curious what these new tests are testing that are not covered by the other tests?  It would probably be helpful to have a more detailed description at part of the doxygen on TEST(ISComm, BufferParse) that provides a more detailed, but still high-level explanation of what this test does and why its necessary.

**Review by kylemallory — APPROVED** [review]

**Review by waltjohnson — APPROVED** [review]
